### PR TITLE
feat: otter sdk training - generate your first sdk specs and command

### PR DESCRIPTION
**Pull request by sdo-1A**
## Proposed change

Otter SDK Training - Generate your first SDK - Specifications step and command step

image
image
image

## Related issues

- :bug: Fixes #(issue)
- :rocket: Feature #(issue)

### CODE REVIEW

File: apps/showcase/src/assets/trainings/sdk/steps/sdk-generation/instructions.html [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        </ul>
+      </li>
+      <li><span class="fa-folder"></span> spec</li>
+    </ul>
+  </li>
+</ul>
+<p>
+  You should also observe these scripts in the <code>package.json</code> file (with the correct project and package names):
+</p>
+<pre class="w-100 bg-body-tertiary px-5">
```
Comment (fpaul-1A) [inline apps/showcase/src/assets/trainings/sdk/steps/sdk-generation/instructions.html:53]: I assume we can only put HTML here, no way to have code highlighting ?

Comment (sdo-1A) [reply apps/showcase/src/assets/trainings/sdk/steps/sdk-generation/instructions.html:53]: indeed, we can only use HTML in these files

**Review by fpaul-1A — APPROVED** [review]

File: apps/showcase/src/assets/trainings/sdk/steps/sdk-generation/instructions.html [hunk 2]
```diff
@@ -0,0 +1,93 @@
+<p>
```
Comment (cpaulve-1A) [inline apps/showcase/src/assets/trainings/sdk/steps/sdk-generation/instructions.html:1]: We can have the generated sdk as a solution for this exercise? I am not sure this is useful, what do the others think?

Comment (fpaul-1A) [reply apps/showcase/src/assets/trainings/sdk/steps/sdk-generation/instructions.html:1]: maybe a link to the `showcase-sdk` that is already versioned would be enough

Comment (sdo-1A) [reply apps/showcase/src/assets/trainings/sdk/steps/sdk-generation/instructions.html:1]: We could display the showcase sdk but there is a risk of the specification file not being aligned with the one in the Swagger Petstore

File: apps/showcase/src/assets/trainings/sdk/steps/sdk-generation/instructions.html [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+</p>
+<p>
+  <i>
+    <b>NOTE:</b> Notice that the <code>yarn spec:upgrade</code> command is equivalent to running the command below, which means that we use the configuration of the
+    <code>openapitools.json</code> file. These configuration properties can also be passed as parameters to the command line (to set or override the property).
+    This tip will be useful for the next steps of the training.
+  </i>
+</p>
+<pre class="w-100 bg-body-tertiary px-5 pre-whitespace">
+  <code>
```
Comment (kpanot) [inline apps/showcase/src/assets/trainings/sdk/steps/sdk-generation/instructions.html:86]: should not be tagged with language?
You can not use `o3r-copy-text-pres` instead?

(same question for other `code`)

Comment (fpaul-1A) [reply apps/showcase/src/assets/trainings/sdk/steps/sdk-generation/instructions.html:86]: this is not possible with the current setup, the html is just used in an `[innerHTML]` with no angular compilation
however, I am currently working on replacing that with `md` files using `ngx-markdown` which include the code highlighting using `prismjs` (check draft PR https://github.com/AmadeusITGroup/otter/pull/2414)
I am also planning on replacing the usage  `o3r-copy-text-pres` to `marked` so we only use one code highlighting technology (the DF demo site also uses PrismJS)

File: apps/showcase/src/assets/trainings/sdk/steps/sdk-generation/instructions.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
+There are several command lines available to generate an SDK depending on the use cases. These cases are covered in the
+<a href="https://github.com/AmadeusITGroup/otter/tree/main/packages/%40ama-sdk/schematics" target="_blank">documentation</a>
+of the package `@ama-sdk/schematics`.
```
Comment (fpaul-1A) [inline apps/showcase/src/assets/trainings/sdk/steps/sdk-generation/instructions.md:3]:
```suggestion
There are several command lines available to generate an SDK depending on the use cases.\
These cases are covered in [@ama-sdk/schematics documentation](https://github.com/AmadeusITGroup/otter/tree/main/packages/%40ama-sdk/schematics).
```

File: apps/showcase/src/assets/trainings/sdk/steps/sdk-specs/instructions.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+The first step in the generation of an SDK is the definition of the specifications. These can be written in a file of type JSON or YAML.
+
+The <a href="https://github.com/AmadeusITGroup/otter/tree/main/packages/%40o3r-training/showcase-sdk" target="_blank">Otter Showcase SDK</a> has been created with the
+<a href="https://github.com/swagger-api/swagger-petstore/blob/master/src/main/resources/openapi.yaml" target="_blank">specifications</a> made available by the
+<a href="https://github.com/swagger-api/swagger-petstore" target="_blank">Swagger Petstore</a> repository.
+You can find these specifications in the `openapi.yml` file of the repository.
+
+### Objective
+Before generating your first SDK using an existing specification file, familiarize yourself with the format of the specifications.
```
Comment (fpaul-1A) [inline apps/showcase/src/assets/trainings/sdk/steps/sdk-specs/instructions.md:9]: Maybe I'm reading it wrong but it almost sounds like a command `familiarize yourself`
```suggestion
The goal of this exercise is to get a basic understanding of the format of the specifications.
```

**Review by fpaul-1A — APPROVED** [review]

File: apps/showcase/src/assets/trainings/sdk/steps/sdk-generation/instructions.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+```bash
+src
+├── api
+├── fixtures
+├── helpers
+├── models
+│   ├── base
+│   ├── core
+│   └── custom
+└── spec
```
Comment (kpanot) [inline apps/showcase/src/assets/trainings/sdk/steps/sdk-generation/instructions.md:31]:
```suggestion
```mermaid
graph LR
    src --> api
    src --> fixtures
    src --> helpers
    src --> models
    src --> spec
    
    models --> base
    models --> core
    models --> custom
```

(or less fancy but not flagged `bash`)

Comment (sdo-1A) [reply apps/showcase/src/assets/trainings/sdk/steps/sdk-generation/instructions.md]: I removed the `bash` flag (I put it there to add some color but keeping it simple is fine too :))

**Review by fpaul-1A — APPROVED** [review]

**Review by kpanot — APPROVED** [review]

**Pull request merged**